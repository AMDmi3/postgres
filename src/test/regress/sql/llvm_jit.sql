--
-- LLVM JIT related tests
--

SET jit_above_cost = 0;
SET jit_inline_above_cost = 0;
SET jit_optimize_above_cost = 0;

CREATE FUNCTION test_thread_local_storage_access()
    RETURNS int4
    AS '/dtmp/postgres/src/test/regress/regress.so' --, 'test_thread_local_storage_access'
    LANGUAGE C;

SELECT test_thread_local_storage_access();
SELECT test_thread_local_storage_access();
SELECT test_thread_local_storage_access();

-- both crash on FreeBSD as boolin() leads to TLS access through
-- isspace() which involves caching locale related data in TLS
SELECT (jsonb_array_elements('[true]'::jsonb)->>0)::boolean;
SELECT (case when random() >= 0.0 then 'true' else 'false' end)::boolean;

SET jit_above_cost to default;
SET jit_inline_above_cost to default;
SET jit_optimize_above_cost to default;

DROP FUNCTION test_thread_local_storage_access;
