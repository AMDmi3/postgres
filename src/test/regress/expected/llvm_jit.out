--
-- LLVM JIT related tests
--
SET jit_above_cost = 0;
SET jit_inline_above_cost = 0;
SET jit_optimize_above_cost = 0;
CREATE FUNCTION test_thread_local_storage_access()
    RETURNS int4
    AS '/dtmp/postgres/src/test/regress/regress.so' --, 'test_thread_local_storage_access'
    LANGUAGE C;
SELECT test_thread_local_storage_access();
 test_thread_local_storage_access 
----------------------------------
                                1
(1 row)

SELECT test_thread_local_storage_access();
 test_thread_local_storage_access 
----------------------------------
                                2
(1 row)

SELECT test_thread_local_storage_access();
 test_thread_local_storage_access 
----------------------------------
                                3
(1 row)

-- both crash on FreeBSD as boolin() leads to TLS access through
-- isspace() which involves caching locale related data in TLS
SELECT (jsonb_array_elements('[true]'::jsonb)->>0)::boolean;
 bool 
------
 t
(1 row)

SELECT (case when random() >= 0.0 then 'true' else 'false' end)::boolean;
 bool 
------
 t
(1 row)

SET jit_above_cost to default;
SET jit_inline_above_cost to default;
SET jit_optimize_above_cost to default;
DROP FUNCTION test_thread_local_storage_access;
